packages/teams-js/src/public/thirdPartyCloudStorage.ts

```diff
@@ -0,0 +1,201 @@
+    fileType: string;
+    /** An array of {@link media.AssembleAttachment | AssembleAttachment} objects representing the media files to be sent as attachment */
+    assembleAttachment: media.AssembleAttachment[];
+  }
+
+  /**
+   * Defines the Callback function received from Third Party App
+   */
+  export interface ThirdPartyAppCallback {
+    /** Callback from third party app */
+    (files: FilesFor3PApps[], error?: SdkError): void;
+  }
+
+  let callback: ThirdPartyAppCallback | null = null;
+
+  /**
+   * Get drag-and-drop files using a callback.
+   *
+   * @param {string} dragAndDropInput - The input related to the drag-and-drop operation.
+   * @param {ThirdPartyAppCallback} thirdPartycallback - callback
+   *   A callback function to handle the result of the operation
+   */
+  export function getDragAndDropFiles(dragAndDropInput: string, thirdPartycallback: ThirdPartyAppCallback): void {
+    if (!thirdPartycallback) {
+      throw new Error('[getDragAndDropFiles] Callback cannot be null');
+    }
+    callback = thirdPartycallback;
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    if (!dragAndDropInput || dragAndDropInput === '') {
+      const invalidInput: SdkError = { errorCode: ErrorCode.INVALID_ARGUMENTS };
+      thirdPartycallback([], invalidInput);
+      return;
+    }
+
+    sendMessageToParent(
+      'thirdPartyCloudStorage.getDragAndDropFiles',
+      [dragAndDropInput],
+      handleGetDragAndDropFilesCallbackRequest,
+    );
+  }
+
+  function handleGetDragAndDropFilesCallbackRequest(fileResult: FileResult): void {
+    if (callback) {
+      if (fileResult && fileResult.error) {
+        callback([], fileResult.error);
+      } else {
+        if (fileResult && fileResult.fileChunk) {
+          try {
+            const assemble: media.AssembleAttachment | null = decodeAttachment(
+              fileResult.fileChunk,
+              fileResult.fileType,
+            );
+            if (assemble) {
+              helper.assembleAttachment.push(assemble);
+            } else {
+              Files3PLogger(
+                `Received a null assemble attachment for when decoding chunk sequence ${fileResult.fileChunk.chunkSequence}; not including the chunk in the assembled file.`,
+              );
+            }
+
+            // we will send the maximum integer as chunkSequence to identify the last chunk
+            if (fileResult.fileChunk.chunkSequence == Number.MAX_SAFE_INTEGER) {
+              const fileBlob = createFile(helper.assembleAttachment, fileResult.fileType);
+
+              if (fileResult.isLastFile && fileBlob) {
+                // Convert blob to File
+                const receivedFile = new File([fileBlob], fileResult.fileName, {
+                  type: fileBlob.type,
+                });
+
+                files.push(receivedFile);
```

Comment:
From TMP we will recieve data in chunks and we will form a file using those chunks.
The callback will be called only once, after the last file is received.
Once we receive the last file we will call this third part callback and send all the files to the third party app.

Reply:
But the files array will still be in memory, so if a second file is dragged in that array will still contain all the chunks from the previous file, right? That seems like a problem

Reply:
When we are receiving the last chunk, then we are converting it into the file and then storing it in files array.
Files array is having array of files. 
When we get the last file then we call the callback provided by 3P app with files array.

Reply:
Yes, and the in memory array still contains those chunks, so if the function is called a second time it will add file chunks to the existing set? Am I misunderstanding something?

Reply:
Modified: I am reassigning the files array to an empty array after we call the callback